ui/src/app/system.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import { WorkflowIcon } from 'lucide-react';
+import '@xyflow/react/dist/style.css';
+import { ThemeToggle } from '@/components/theme-toggle';
+import { getSystem, listPipelines } from '@/services/api';
+import { Badge } from '@/components/ui/badge';
+import { Pipeline as PipelineComponent } from '@/app/pipeline';
+import { System } from '@/types/system';
+import { Pipeline } from '@/types/pipeline';
+import { ReactFlowProvider } from '@xyflow/react';
+
+export default function System() {
+  const [system, setSystem] = useState<System>();
+  const [pipelines, setPipelines] = useState<Pipeline[]>();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setSystem(await getSystem());
+    };
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setPipelines(await listPipelines());
+    };
+    fetchData();
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    const fetchData = async () => {
      setSystem(await getSystem());
      setPipelines(await listPipelines());
    };
    fetchData();
  }, []);
```

could we combine these two? is this where we need to do usecallback?